app/src/main/java/com/example/lister/MainActivity.java

```diff
@@ -229,8 +233,9 @@ protected String doInBackground(Object... params) {
         protected void onPostExecute(String result) {
             MainActivity activity = mActivityWeakReference.get();
             if (activity != null && !activity.isFinishing()) {
-                //Todo: Do something with the result.
                 Log.d("CloudVision", "Success! " + result);
+                helper = ListDatabaseHelper.getInstance(itemContext);
+                helper.updateItemPrice(getPrice(result), itemIdForPhoto);
```

Comment:
Does this automatically update the cost in the list without like going back and reopoing it? If it doesn't I think there should be a way but it might not be that important now I guess.

Reply:
I tied onResume(), but on further testing it doesn't yet. I'll check it out again.

Reply:
you will probably have to call a method that's like `notifydatasetchanged` or something on the adapter, but that's only if changing the data in the adapter's list doesn't make it update automatically. so you might be able to just update that specific item or just get an entirely new list of all the list items. worst case maybe we just recreate the fragment? 

imo automatically showing the data changing is more important than efficiency rn

Reply:
unfortunately, because parsing is done in the background, it seems that any changing of text will have to be done by the activity rather than adapter. Finding the current fragment id seems to return null so can't call a method to refresh the fragment, and onResume() recreating the fragment results in a loop. Seeing if there's a way to use onWindowFocusChanged() outside of an activity but so far nothing